library/core/tests/iter/traits/iterator.rs

```diff
@@ -497,6 +497,52 @@ fn test_collect() {
     assert!(a == b);
 }
 
+#[test]
+fn test_try_collect() {
+    use core::ops::ControlFlow::{Break, Continue};
+
+    let u = vec![Some(1), Some(2), Some(3)];
+    let v = u.into_iter().try_collect::<Vec<i32>>();
+    assert_eq!(v, Some(vec![1, 2, 3]));
+
+    let u = vec![Some(1), Some(2), None, Some(3)];
+    let mut it = u.into_iter();
+    let v = it.try_collect::<Vec<i32>>();
+    assert_eq!(v, None);
+    let v = it.try_collect::<Vec<i32>>();
+    assert_eq!(v, Some(vec![3]));
+
+    let u: Vec<Result<i32, ()>> = vec![Ok(1), Ok(2), Ok(3)];
+    let v = u.into_iter().try_collect::<Vec<i32>>();
+    assert_eq!(v, Ok(vec![1, 2, 3]));
```

Comment:
This is already possible with `.collect()`. Thanks to `impl<A, E, V> FromIterator<Result<A, E>> for Result<V, E> where V: FromIterator<A>`.

Reply:
Right, but the main benefits of `try_collect()` are increasing the discoverability of this technique and allowing `Try`-but-not-`FromIterator` types (e.g. `ControlFlow`) to be fallibly collected as well. See my comment on the tracking issue [here](https://github.com/rust-lang/rust/issues/94047#issuecomment-1042094358) for a more thorough description of the differences between `try_collect()` and `collect()`.